### PR TITLE
Heartbeat and Flow Control

### PR DESCRIPTION
**Pull request by scottf**
Also contains JsMulti tool queue fix

### CODE REVIEW

File: src/main/java/io/nats/client/api/ConsumerConfiguration.java [hunk 1]
```diff
@@ -420,25 +450,48 @@ public Builder maxAckPending(long maxAckPending) {
...
+                    ackWait == null ? Duration.ofSeconds(30) : ackWait,
+                    maxDeliver,
+                    filterSubject,
+                    replayPolicy == null ? ReplayPolicy.Instant : replayPolicy,
+                    sampleFrequency,
+                    rateLimit,
+                    deliverSubject,
+                    maxAckPending,
+                    idleHeartbeat == null ? Duration.ZERO : idleHeartbeat,
+                    flowControl
```
Comment (scottf) [inline src/main/java/io/nats/client/api/ConsumerConfiguration.java:494]: validations / defaults

File: src/examples/java/io/nats/examples/jsmulti/Arguments.java [hunk 2]
```diff
@@ -40,6 +41,8 @@
     AckPolicy ackPolicy = AckPolicy.Explicit;
     int ackFrequency = 1;
     int batchSize = 10;
+    final String queueDurable = "qd" + uniqueEnough();
+    final String queueName = "qn" + uniqueEnough();
```
Comment (scottf) [inline src/examples/java/io/nats/examples/jsmulti/Arguments.java:45]: need to use the same durable and queue name for every thread

File: src/test/java/io/nats/client/support/ValidatorTests.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public static void beforeAll() {
 
     @Test
     public void testValidateSubject() {
-        allowedRequired(Validator::validateSubject, Arrays.asList(PLAIN, HAS_DOT, HAS_STAR, HAS_GT, HAS_DOLLAR));
+        allowedRequired(Validator::validateSubject, Arrays.asList(PLAIN, HAS_PRINTABLE, HAS_DOT, HAS_STAR, HAS_GT, HAS_DOLLAR));
```
Comment (scottf) [inline src/test/java/io/nats/client/support/ValidatorTests.java:42]: one more piece of data in the tests

**Review by ColinSullivan1 — APPROVED** [review]
LGTM.

**Pull request merged**